back-end/helpers/intent_analyzer.js

```diff
@@ -0,0 +1,38 @@
+export default class IntentAnalyzer{
+    analyzeIntents(models) {
+      let newIntents = [];
+
+      for (let model in models) {
+        newIntents.push(this.formatModel(model))
```

Comment:
Both of these should be const. This is because each time we declare the variable, we never reassign it. We mutate its contents, not what it is representing.